cupyx/_pinned_array.py

```diff
@@ -0,0 +1,157 @@
+import numpy
+
+from cupy import cuda
+from cupy._creation.basic import _new_like_order_and_strides
+from cupy.core import internal
+
+
+def _update_shape(a, shape):
+    if shape is None and a is not None:
+        shape = a.shape
+    elif isinstance(shape, int):
+        shape = (shape,)
+    else:
+        shape = tuple(shape)
+    return shape
+
+
+def _update_order_char(x, order_char):
+    # This is a pure Python version of cupy.core.core._update_order_char()
+    # update order_char based on array contiguity
+    if order_char == ord('A'):
+        if x.flags.f_contiguous:
+            order_char = ord('F')
+        else:
+            order_char = ord('C')
+    elif order_char == ord('K'):
+        if x.flags.f_contiguous:
+            order_char = ord('F')
+        elif x.flags.c_contiguous:
+            order_char = ord('C')
+    return order_char
+
+
+def empty_pinned(shape, dtype=float, order='C'):
+    """Returns an array without initializing the elements.
+
+    This is a convenience function which is just :func:`numpy.empty`,
+    except that the underlying memory is allocated from CuPy's pinned
+    memory pool.
```

Comment:
Users may disable the memory pool, so:

```suggestion
    except that the underlying memory is allocated as a pinned memory.
```

Reply:
Done in 1bc4e3c with minor changes.